LocalPubSubSpec: A pub-sub topic running locally shouldn't publish after unsubscribe

Only seen on scala3 so far

https://github.com/akka/akka/runs/3161613823?check_suite_focus=true

```
[info] - must doesn't publish after unsubscribe *** FAILED *** (119 milliseconds)
[info]   java.lang.AssertionError: Received 1 excess messages on LoggingTestKitImpl(1,Some(DEBUG),None,None,Some(Topic list updated),None,None,Map(),None).
[info]   at akka.actor.testkit.typed.internal.LoggingTestKitImpl.expect(LoggingTestKitImpl.scala:103)
[info]   at akka.actor.typed.pubsub.LocalPubSubSpec.f$proxy4$3(LocalPubSubSpec.scala:131)
[info]   at akka.actor.typed.pubsub.LocalPubSubSpec.$init$$$anonfun$5$$anonfun$4(LocalPubSubSpec.scala:127)
```

--- a/akka-actor-testkit-typed/src/main/mima-filters/2.6.15.backwards.excludes/excess-checking.excludes
+++ b/akka-actor-testkit-typed/src/main/mima-filters/2.6.15.backwards.excludes/excess-checking.excludes
@@ -1,0 +1,6 @@
+# Extra methods on interfaces not for user extension:
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.testkit.typed.javadsl.LoggingTestKit.withCheckExcess")
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.testkit.typed.scaladsl.LoggingTestKit.withCheckExcess")
+
+# Impl
+ProblemFilters.exclude[Problem]("akka.actor.testkit.typed.internal.LoggingTestKitImpl.*")

--- a/akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/LoggingTestKitImpl.scala
+++ b/akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/LoggingTestKitImpl.scala
@@ -24,7 +24,8 @@ import akka.testkit.TestKit
  * INTERNAL API
  */
 @InternalApi private[akka] object LoggingTestKitImpl {
-  def empty: LoggingTestKitImpl = new LoggingTestKitImpl(1, None, None, None, None, None, None, Map.empty, None)
+  def empty: LoggingTestKitImpl =
+    new LoggingTestKitImpl(1, None, None, None, None, None, None, Map.empty, checkExcess = true, None)
 }
 
 /**
@@ -39,6 +40,7 @@ import akka.testkit.TestKit
     messageRegex: Option[Regex],
     cause: Option[Class[_ <: Throwable]],
     mdc: Map[String, String],
+    checkExcess: Boolean,
     custom: Option[Function[LoggingEvent, Boolean]])
     extends javadsl.LoggingTestKit
     with scaladsl.LoggingTestKit {
@@ -92,14 +94,14 @@ import akka.testkit.TestKit
     try {
       val result = code
 
-      // wait some more when occurrences=0 to find asynchronous exceess messages
+      // wait some more when occurrences=0 to find asynchronous excess messages
       if (occurrences == 0)
         awaitNoExcess(settings.ExpectNoMessageDefaultTimeout)
 
       if (!awaitDone(settings.FilterLeeway))
         if (todo > 0)
           throw new AssertionError(s"Timeout (${settings.FilterLeeway}) waiting for $todo messages on $this.")
-        else
+        else if (checkExcess)
           throw new AssertionError(s"Received ${-todo} excess messages on $this.")
       result
     } finally {
@@ -151,6 +153,9 @@ import akka.testkit.TestKit
     import akka.util.ccompat.JavaConverters._
     withMdc(newMdc.asScala.toMap)
   }
+
+  override def withCheckExcess(check: Boolean): LoggingTestKitImpl =
+    copy(checkExcess = check)
 
   override def withCustom(newCustom: Function[LoggingEvent, Boolean]): LoggingTestKitImpl =
     copy(custom = Option(newCustom))

--- a/akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/LoggingTestKit.scala
+++ b/akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/LoggingTestKit.scala
@@ -77,6 +77,11 @@ import akka.annotation.DoNotInherit
   def withMdc(newMdc: java.util.Map[String, String]): LoggingTestKit
 
   /**
+   * After matching the expected number of hits, check for excess messages
+   */
+  def withCheckExcess(check: Boolean): LoggingTestKit
+
+  /**
    * Matching events for which the supplied function returns `true`.
    */
   def withCustom(newCustom: Function[LoggingEvent, Boolean]): LoggingTestKit

--- a/akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/LoggingTestKit.scala
+++ b/akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/LoggingTestKit.scala
@@ -77,6 +77,11 @@ import akka.annotation.DoNotInherit
   def withMdc(newMdc: Map[String, String]): LoggingTestKit
 
   /**
+   * After matching the expected number of hits, check for excess messages
+   */
+  def withCheckExcess(checkExcess: Boolean): LoggingTestKit
+
+  /**
    * Matching events for which the supplied function returns`true`.
    */
   def withCustom(newCustom: Function[LoggingEvent, Boolean]): LoggingTestKit

--- a/akka-actor-typed-tests/src/test/scala/akka/actor/typed/MailboxSelectorSpec.scala
+++ b/akka-actor-typed-tests/src/test/scala/akka/actor/typed/MailboxSelectorSpec.scala
@@ -74,7 +74,7 @@ class MailboxSelectorSpec extends ScalaTestWithActorTestKit("""
       }, MailboxSelector.bounded(2))
       actor ! "one" // actor will block here
       actor ! "two"
-      LoggingTestKit.deadLetters().expect {
+      LoggingTestKit.deadLetters().withCheckExcess(false).expect {
         // one or both of these doesn't fit in mailbox
         // depending on race with how fast actor consumes
         actor ! "three"

--- a/akka-actor-typed-tests/src/test/scala/akka/actor/typed/pubsub/LocalPubSubSpec.scala
+++ b/akka-actor-typed-tests/src/test/scala/akka/actor/typed/pubsub/LocalPubSubSpec.scala
@@ -19,7 +19,7 @@ class LocalPubSubSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike wit
 
     "publish to all local subscriber actors of a topic" in {
       val fruitTopic =
-        LoggingTestKit.debug("Topic list updated").expect {
+        LoggingTestKit.debug("Topic list updated").withCheckExcess(false).expect {
           testKit.spawn(Topic[String]("fruit"))
         }
 
@@ -50,9 +50,10 @@ class LocalPubSubSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike wit
     }
 
     "publish to all subscriber actors across several instances of the same topic" in {
-      val (fruitTopic1, fruitTopic2) = LoggingTestKit.debug("Topic list updated").withOccurrences(2).expect {
-        (testKit.spawn(Topic[String]("fruit")), testKit.spawn(Topic[String]("fruit")))
-      }
+      val (fruitTopic1, fruitTopic2) =
+        LoggingTestKit.debug("Topic list updated").withOccurrences(2).withCheckExcess(false).expect {
+          (testKit.spawn(Topic[String]("fruit")), testKit.spawn(Topic[String]("fruit")))
+        }
 
       try {
         val probe1 = testKit.createTestProbe[String]()
@@ -126,7 +127,7 @@ class LocalPubSubSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike wit
 
     "doesn't publish after unsubscribe" in {
       val fruitTopic =
-        LoggingTestKit.debug("Topic list updated").expect {
+        LoggingTestKit.debug("Topic list updated").withCheckExcess(false).expect {
           testKit.spawn(Topic[String]("fruit"))
         }
 

